### ISSUE TITLE
VectorIndex data not updated properly for multidimensional ragged array columns

### ISSUE THREAD

**Issue by cechava** [NeurodataWithoutBorders/matnwb#390]
```matlab
% Create file
file= NwbFile( ...
    'session_start_time', '2021-01-01 00:00:00', ...
    'identifier', 'ident1', ...
    'session_description', 'test_file' ...
    );

% Define Vector Data Objects with first row of table
start_time_exp = types.hdmf_common.VectorData( ...
    'description', 'start times column', ...
    'data', types.untyped.DataPipe( ...
        'data', 1, ...  
        'maxSize', Inf ...
    ) ...
);
stop_time_exp = types.hdmf_common.VectorData( ...
    'description', 'stop times column', ...
    'data', types.untyped.DataPipe( ...
        'data', 10, ...  
        'maxSize', Inf ...
    ) ...
);
random_exp = types.hdmf_common.VectorData( ...
    'description', 'random data column', ...
    'data', types.untyped.DataPipe( ...
        'data', rand(3,2,5), ...  #random data
        'maxSize', [3, 2, Inf], ...
        'axis', 3 ...
    ) ...
);
random_exp_index = types.hdmf_common.VectorIndex( ...
    'description', 'index to random data column', ...
     'target',types.untyped.ObjectView(random_exp), ...
    'data', types.untyped.DataPipe( ...
        'data', [5], ...  
        'maxSize', [Inf, 1], ...
        'axis', 1 ...
    ) ...
);
ids_exp = types.hdmf_common.ElementIdentifiers( ...
    'data', types.untyped.DataPipe( ...
        'data', 0, ...  # data must be numerical
        'maxSize', Inf ...
    ) ...
);
% Create expandable table
colnames = {'start_time', 'stop_time', 'randomvalues'};
file.intervals_trials = types.core.TimeIntervals( ...
    'description', 'test expdandable dynamic table', ...
    'colnames', colnames, ...
    'start_time', start_time_exp, ...
    'stop_time', stop_time_exp, ...
    'randomvalues', random_exp, ...
    'randomvalues_index', random_exp_index, ...
    'id', ids_exp ...
);  
% Export file
nwbExport(file, 'multiRaggedExpandableTableTest.nwb');
% Read in file
read_file = nwbRead('multiRaggedExpandableTableTest.nwb');
% add individual rows
read_file.intervals_trials.addRow( ...
    'start_time', 2, ...
    'stop_time', 20, ...
    'randomvalues', rand(3,2,6), ...
    'id', 1 ...
);
```

Output after `addRow` 
```matlab

>> read_file.intervals_trials.vectordata.get('randomvalues').data.internal

ans = 

  BoundPipe with properties:

            config: [1×1 types.untyped.datapipe.Configuration]
    pipeProperties: {1×2 cell}
              stub: [1×1 types.untyped.DataStub]
              axis: 3
            offset: 11
          dataType: 'double'
           maxSize: [3 2 Inf]
              dims: [3 2 11]
          filename: 'multiRaggedExpandableTableTest.nwb'
              path: '/intervals/trials/randomvalues'

read_file.intervals_trials.vectordata.get('randomvalues_index').data(:)

ans =

     5
     8
``` 

Second entry of `randomvalues_index` data should be 11. 

No issues for 1D ragged array columns.

**Comment by ln-vidrio** [comment]
I think this is just jumping the gun a little bit, we haven't swapped dimensions yet so `addRow` will still be using the first dimension to concatenate.

**Comment by ln-vidrio** [comment]
Yes `addRow` currently assumes that you are appending on the first dimension which will be fixed in #387.

**Comment by cechava** [comment]
Thanks for clarifying
